api/scpca_portal/models/job.py

```diff
@@ -104,6 +111,27 @@ def get_sample_job(
             },
         )
 
+    @classmethod
+    def get_retry_jobs(cls) -> List[Self] | None:
+        """
+        Prepare new unsaved Job instances to retry terminated jobs.
+        Exclude those with critical_error is True and retry_on_termination is False.
+        Set each instance's attempt to the base terminated job's attempt incremented by 1.
+        Return the new instances, otherwise None if no terminated jobs for retry available.
+        """
+        if terminated_jobs := Job.objects.filter(
+            state=JobStates.TERMINATED.name, retry_on_termination=True
+        ).exclude(critical_error=True):
```

Comment:
I've excluded terminated jobs with `critical_error` set to `True`. Should we also apply this logic in the `Job::get_retry_job` instance method?